### PR TITLE
Reactor

### PR DESCRIPTION
**Pull request by martinsmid**
Hi, this is www.reactor.am 's integration. I hope everything is ok with it.

### CODE REVIEW

**Comment by f2prateek** [comment]
Hey @martinsmid ,

Thanks for submitting a pull request. Due to a high number of pull requests across different Segment repositories we require a bit more information from partners wishing to join the platform.

Could you please complete a [business survey](https://segment.com/partners/businessqs) and a [technical survey](
https://segment.com/partners/techqs). The business survey helps us understand what your service is, and how to communicate that to our customers. The technical survey helps us better plan supporting the integration, and how you should contribute. 

We'll review these surveys within several days, and if all is okay, we'll reopen the pull request for review.

**Comment by martinsmid** [comment]
Hi,

the technical and business surveys have already been filled in few months
ago.
Technical
https://docs.google.com/a/reactor.am/forms/d/1rAOchI9uNXlcw6eBm8DIYvxE5wRZCL1c12yFxV6AV7Y/formResponse?edit2=2_ABaOnufLakRe87G45Lfkj_9Olai-ey50cgjvr8GLwCBvBVjqdc1z4dCBcsAnd2c
Business
https://docs.google.com/forms/d/14Z5AMsUNc1jQXYwide43w8KbGIghI8LtfVMN-KDJTlY/formResponse?edit2=2_ABaOnue7UlE7JJfe6qHbEqZT0Q5qaaoIC0tcRjI96jQcw-rHqGHFRaCNpsTq9dY


2015-02-05 2:32 GMT+01:00 Prateek Srivastava <notifications@github.com>:

**Comment by f2prateek** [comment]
@martinsmid Thanks! I'll let our partners team know that your integration is ready to go and we'll start a PR review once they approve it.

File: lib/reactor/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    .readyOnLoad()
+    .global('ReactorAmObject')
+    .global('_rcr')
+    .option('apikey', null)
+    .tag('collector', '<script src="//www.reactor.am/static/collector.js">');
+
+
+Reactor.prototype.initialize = function(page) {
+    (function(w,d,r){
+        w['ReactorAmObject']=r;w[r]=w[r]||function(){(w[r].q=w[r].q||[]).push(arguments)};
```
Comment (f2prateek) [inline lib/reactor/index.js]: You can un-minify this line of code.

**Comment by f2prateek** [comment]
@martinsmid Is there any particular reason why you need an analytics.js integration over a server side integration? Unless there is a compelling reason, the latter would expand your reach to a wider user base.

**Comment by martinsmid** [comment]
@f2prateek Updated the code with latest master. The code seems to fail on build.js, which is generated and is outside git.
We would like to finish this integration. After successfully integrated, we will consider the possibilities/limitations of server side integration.

**Comment by f2prateek** [comment]
For the test failure, you have to change the integrations count to 81 (from 80) in https://github.com/segmentio/analytics.js-integrations/blob/ca6e938655f501e3826a7d548250fc6d0d034b12/test/index.js#L24

File: lib/reactor/index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * Module dependencies.
+ */
+
+var integration = require('analytics.js-integration');
+var load = require('load-script');
+
+module.exports = function(analytics){
```
Comment (f2prateek) [inline lib/reactor/index.js]: @martinsmid What is this needed for?

File: lib/reactor/index.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        page: path(props),
+        url: props.url
+    });
+}
+
+Reactor.prototype.loaded = function(){
+    return !!window._rcr;
+};
+
+function path(properties){
```
Comment (f2prateek) [inline lib/reactor/index.js]: This doesn't need to be it's own method, you can inline it since it's only use in page.

File: lib/reactor/index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            case "boolean":
+                reactor_prop_name += '_b';
+                prop_value = 0 + prop_value;
+                break;
+            case "object":
+                if (prop_value === null) {
+                    reactor_prop_name += '_s';
+                    prop_value = null;
+                } else
+                    continue;
```
Comment (f2prateek) [inline lib/reactor/index.js]: The code formatting obscures the purpose. Could you wrap this in braces?

Comment (f2prateek) [inline lib/reactor/index.js]: Also why would you need to continue here - looks like you accept null values but ignore non-nulls?

File: lib/reactor/index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                break;
+            default:
+                continue;
+                break;
```
Comment (f2prateek) [inline lib/reactor/index.js]: Continue followed by break? This doesn't look right.

**Comment by f2prateek** [comment]
@martinsmid There's still a couple of changes before this PR can be merged. I've made some inline comments, but additionally:

1. Our project use 2 spaces, not 4 spaces as in this formatting.
2.  I'm to sure why ` test/tests.js` is formatted for you, but make sure you don't change it in the PR

The tests are also nested under `Universal` tag, you shouldn't need that.

This is an example of a PR of how the end result should look like https://github.com/segmentio/analytics.js-integrations/pull/596/files

**Comment by ndhoule** [comment]
This PR has gone stale so I'm closing this. If you'd like to get this going again, please reach out to our partners team at partners@segment.com. Thanks!

**Pull request closed without merging**